lib/mpl_toolkits/mplot3d/proj3d.py

```diff
@@ -173,19 +173,18 @@ def _ortho_transformation(zfront, zback):
 def _proj_transform_vec(vec, M):
     vecw = np.dot(M, vec)
     w = vecw[3]
-    # clip here..
     txs, tys, tzs = vecw[0]/w, vecw[1]/w, vecw[2]/w
     return txs, tys, tzs
 
 
-def _proj_transform_vec_clip(vec, M):
+def _proj_transform_vec_clip(vec, M, focal_length):
     vecw = np.dot(M, vec)
     w = vecw[3]
-    # clip here.
     txs, tys, tzs = vecw[0] / w, vecw[1] / w, vecw[2] / w
-    tis = (0 <= vecw[0]) & (vecw[0] <= 1) & (0 <= vecw[1]) & (vecw[1] <= 1)
-    if np.any(tis):
-        tis = vecw[1] < 1
+    if np.isinf(focal_length):  # don't clip orthographic projection
+        tis = np.ones(txs.shape, dtype=bool)
+    else:
+        tis = (-1 <= txs) & (txs <= 1) & (-1 <= tys) & (tys <= 1) & (tzs <= 0)
```

Comment:
Note that we may want to switch to tzs <= -1 in the future, since a value of 0 means that there is no clipping between the image plane and the camera as in this diagram. But at the moment I thought that the view all the way up to the camera was most appropriate.
![image](https://github.com/matplotlib/matplotlib/assets/14363975/af7cf499-b1d3-457c-a705-5d3b1179d6e5)